src/sql/src/query_model/hir/qgm_from_hir.rs

```diff
@@ -60,6 +60,25 @@ impl FromHir {
     /// Generates a sub-graph representing the given expression.
     fn generate_internal(&mut self, expr: HirRelationExpr) -> Result<BoxId, QGMError> {
         match expr {
+            HirRelationExpr::Constant { rows, typ } => {
+                if typ.arity() == 0 || rows.len() != 1 {
```

Comment:
Why is creating the values box disallowed when rows.len() == 1 even when the arity is 0?

Reply:
I wanted to say that we only want to support a `Constant` which is a singleton collection consisting of an empty tuple, so the check should actually be `typ.arity() == 0 && rows.len() == 1`